### PR TITLE
NTRN-389 audit modules fixes

### PR DESCRIPTION
**Pull request by quasisamurai**
[Test run](https://github.com/neutron-org/neutron-tests/actions/runs/4341210957/jobs/7580624365)

[TASK](https://p2pvalidator.atlassian.net/browse/NTRN-389)

### CODE REVIEW

File: x/interchainqueries/types/message_remove_interchain_query.go [hunk 1]
```diff
@@ -7,24 +7,22 @@ import (
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
-var _ sdk.Msg = &MsgRemoveInterchainQueryRequest{}
```
Comment (mmikev) [inline x/interchainqueries/types/message_remove_interchain_query.go:10]: Why is this removed?
It's pretty necessary compile check that `MsgRemoveInterchainQueryRequest` really implements sdk.Msg

File: x/interchainqueries/keeper/keeper.go [hunk 2]
```diff
@@ -284,7 +285,8 @@ func (k Keeper) GetClientState(ctx sdk.Context, clientID string) (*tendermintLig
 func (k *Keeper) CollectDeposit(ctx sdk.Context, queryInfo types.RegisteredQuery) error {
 	owner, err := queryInfo.GetOwnerAddress()
 	if err != nil {
-		panic(err.Error())
+		ctx.Logger().Debug("failed to get owner address from given query", "query_info", queryInfo)
```
Comment (albertandrejev) [inline x/interchainqueries/keeper/keeper.go]: Do you know scenario where Owner will not be valid Address? I think that if we will simply write message to log it may lead to data inconsistency, problem hiding, and this will start growing. Network halting is not the best solution of course, but at least in this case we will see this weird situation if we will catch unexpected error like this.

File: x/interchaintxs/types/tx.go [hunk 3]
```diff
@@ -10,8 +10,6 @@ import (
 	"github.com/gogo/protobuf/proto"
 )
 
-var _ codectypes.UnpackInterfacesMessage = MsgSubmitTx{}
```
Comment (albertandrejev) [inline x/interchaintxs/types/tx.go]: Same question as @pr0n00gler

File: x/interchainqueries/types/message_remove_interchain_query.go (hunk 1 above)
Comment (quasisamurai) [reply x/interchainqueries/types/message_remove_interchain_query.go:10]: that happened accidentally, tnx

**Review by zavgorodnii — DISMISSED** [review]

**Review by zavgorodnii — APPROVED** [review]

**Review by albertandrejev — APPROVED** [review]
